From bcbec1b534d00138df17e0f76b80bdd25e5f459e Mon Sep 17 00:00:00 2001
From: Vanessa Maegima <vanessa.maegima@foundries.io>
Date: Tue, 29 Mar 2022 13:44:56 -0300
Subject: [PATCH 140/233] [FIO internal] arch: arm: dts: imx8mp-evk: enable
 i2c5 bus

Enable the I2C5 bus in U-Boot and SPL dts. This is needed to enable
the SE050 support in this bus.

Flexcan1 conflicts with I2C5 pins, so disable it.

Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>
Co-developed-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 arch/arm/dts/imx8mp-evk-u-boot.dtsi | 24 ++++++++++++++++++++++++
 arch/arm/dts/imx8mp-evk.dts         | 25 +++++++++++++++++++++++++
 2 files changed, 49 insertions(+)

diff --git a/arch/arm/dts/imx8mp-evk-u-boot.dtsi b/arch/arm/dts/imx8mp-evk-u-boot.dtsi
index 203d05e66ea..7661926a5c0 100644
--- a/arch/arm/dts/imx8mp-evk-u-boot.dtsi
+++ b/arch/arm/dts/imx8mp-evk-u-boot.dtsi
@@ -144,6 +144,10 @@
 	u-boot,dm-spl;
 };
 
+&i2c5 {
+	u-boot,dm-spl;
+};
+
 &pinctrl_i2c1 {
 	u-boot,dm-spl;
 };
@@ -152,6 +156,22 @@
 	u-boot,dm-spl;
 };
 
+&pinctrl_i2c3 {
+	u-boot,dm-spl;
+};
+
+&pinctrl_i2c3_gpio {
+	u-boot,dm-spl;
+};
+
+&pinctrl_i2c5 {
+	u-boot,dm-spl;
+};
+
+&pinctrl_i2c5_gpio {
+	u-boot,dm-spl;
+};
+
 &usdhc1 {
 	u-boot,dm-spl;
 	assigned-clocks = <&clk IMX8MP_CLK_USDHC1>;
@@ -189,6 +209,10 @@
 	u-boot,dm-spl;
 };
 
+&{/soc@0/bus@30800000/i2c@30a40000/gpio@20} {
+	u-boot,dm-spl;
+};
+
 &pinctrl_pmic {
 	u-boot,dm-spl;
 };
diff --git a/arch/arm/dts/imx8mp-evk.dts b/arch/arm/dts/imx8mp-evk.dts
index 0df546dfd06..6f9579d10af 100644
--- a/arch/arm/dts/imx8mp-evk.dts
+++ b/arch/arm/dts/imx8mp-evk.dts
@@ -379,6 +379,17 @@
 	};
 };
 
+&i2c5 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c5>;
+	pinctrl-1 = <&pinctrl_i2c5_gpio>;
+	scl-gpios = <&gpio5 3 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio5 4 GPIO_ACTIVE_HIGH>;
+	gpio = <&pca6416 2 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 &lcdif1 {
 	status = "okay";
 };
@@ -593,6 +604,13 @@
 		>;
 	};
 
+	pinctrl_i2c5: i2c5grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SPDIF_TX__I2C5_SCL			0x400001c3
+			MX8MP_IOMUXC_SPDIF_RX__I2C5_SDA			0x400001c3
+		>;
+	};
+
 	pinctrl_i2c1_gpio: i2c1grp-gpio {
 		fsl,pins = <
 			MX8MP_IOMUXC_I2C1_SCL__GPIO5_IO14        	0x1c3
@@ -614,6 +632,13 @@
 		>;
 	};
 
+	pinctrl_i2c5_gpio: i2c5grp-gpio {
+		fsl,pins = <
+			MX8MP_IOMUXC_SPDIF_TX__GPIO5_IO03        	0x1c3
+			MX8MP_IOMUXC_SPDIF_RX__GPIO5_IO04        	0x1c3
+		>;
+	};
+
 	pinctrl_mipi_dsi_en: mipi_dsi_en {
 		fsl,pins = <
 			MX8MP_IOMUXC_GPIO1_IO08__GPIO1_IO08	0x16
-- 
2.25.1

